src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -648,6 +658,50 @@ private async Task ExecuteUpgradeNuGetProjectCommandAsync(object sender, EventAr
             }
         }
 
+        private void ExecuteUpgradeProjectJsonNuGetProjectCommand(object sender, EventArgs e)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await ExecuteUpgradeProjectJsonNuGetProjectCommandAsync(sender, e);
+            })
+           .PostOnFailure(nameof(NuGetPackage), nameof(ExecuteUpgradeNuGetProjectCommand));
+        }
+
+        private async Task ExecuteUpgradeProjectJsonNuGetProjectCommandAsync(object sender, EventArgs e)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            if (ShouldInitializeSolutionExperiences())
+            {
+                await InitializeSolutionExperiencesAsync();
+            }
+
+            Project project = VsMonitorSelection.GetActiveProject();
+
+            string uniqueName = await project.GetCustomUniqueNameAsync();
+            NuGetProject nuGetProject = await SolutionManager.Value.GetNuGetProjectAsync(uniqueName);
+
+            if (await NuGetProjectUpgradeUtility.IsNuGetProjectUpgradeableAsync(nuGetProject as ProjectJsonNuGetProject))
```

Comment:
```suggestion
            if (!await NuGetProjectUpgradeUtility.IsNuGetProjectUpgradeableAsync(nuGetProject as ProjectJsonNuGetProject))
```

Is this what you mean here?

Reply:
I'm missing a commit here. 

Nice catch.

I still even have my TODO there.